src/Tes.Runner/Transfer/BlobOperationPipeline.cs

```diff
@@ -79,4 +80,22 @@ protected async Task<long> ExecutePipelineAsync(List<BlobOperationInfo> operatio
 
         return bytesProcessed;
     }
+
+    protected static async Task WhenAllFailFast(IEnumerable<Task> tasks)
+    {
+        var taskList = tasks.ToList();
+        while (taskList.Count > 0)
+        {
+            var completedTask = await Task.WhenAny(taskList);
+            if (completedTask.IsFaulted)
+            {
+                throw completedTask.Exception!.InnerException!;
```

Comment:
Just to future-proof this. I'd suggest `throw completedTask.Exception!?.InnerException!;`